server/container_create_linux.go

```diff
@@ -343,9 +343,16 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrIface.Contai
 			}
 
 			specgen.SetProcessOOMScoreAdj(int(resources.OomScoreAdj))
-			specgen.SetLinuxResourcesCPUCpus(resources.CPUsetCPUs)
 			specgen.SetLinuxResourcesCPUMems(resources.CPUsetMems)
 
+			specgen.SetLinuxResourcesCPUCpus(resources.CPUsetCPUs)
+			if resources.CPUsetCPUs == "" {
+				infraCPUs := s.Runtime().GetInfraCPUs()
+				if infraCPUs != nil {
+					specgen.SetLinuxResourcesCPUCpus(infraCPUs.String())
+				}
```

Comment:
we don't want to set this here I believe. what this is doing is settting a new, non-infra container to use infra cpus, if it has cpus set. What we should do is set infra containers to be in those cpus (in server/sandbox_run_linux.go). Or, we can move the infra container to the cpus when the container is created (similar to how a container is in container_update_resources)

does that make sense?

Reply:
thinking about this more, we should unconditionally set the infra container's cpuset if `infra-ctr-cpus` is set. we'll have to do that in `server/sandbox_run_linux.go`

Reply:
done